contracts/utils/cryptography/ECDSA.sol

```diff
@@ -52,17 +52,39 @@ library ECDSA {
                 v := add(shr(255, vs), 27)
             }
         } else {
-            revert("ECDSA: invalid signature length");
+            return (address(0), "ECDSA: invalid signature length");
         }
 
-        return recover(hash, v, r, s);
+        return (recover(hash, v, r, s), "");
```

Comment:
maybe that should call `tryRecover(hash, v,r,s)` instead 
allocating empty string has a cost